.github/workflows/core-network-services-deployment.yml

```diff
@@ -0,0 +1,55 @@
+---
+name: core-network-services-deployment
+
+on: 
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'terraform/environments/core-network-services/**'
+  workflow_dispatch:
+
+env:
+  AWS_ACCESS_KEY_ID:  ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY:  ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  TF_IN_AUTOMATION: true
+
+jobs:
+  core-network-services-deployment-plan:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Terraform plan
+        run: |
+          chmod u+x scripts/terraform_plan_comment_update.sh
+          scripts/terraform_plan_comment_update.sh terraform/environments/core-network-services
+        env:
+          SECRET: ${GITHUB_TOKEN}
+          GITHUB_REPOSITORY: ${GITHUB_REPOSITORY}
+          PULL_REQUEST_NUMBER: ${{ github.event.pull_request.number }}
+
+  core-network-services-deployment-real-apply:
+    runs-on: ubuntu-latest
+    if: github.event.ref == 'refs/heads/main'
```

Comment:
You can use `if: ${{ github.event.pull_request.merged == 'true' }}` to check if the PR has been merged into main, rather than checking the `ref`, if easier.

Reply:
updated